matchzoo/preprocessors/mvlstm_preprocessor.py

```diff
@@ -0,0 +1,103 @@
+"""MVLSTM Preprocessor."""
+
+import logging
+
+from tqdm import tqdm
+
+from matchzoo import engine, processor_units
+from matchzoo import DataPack
+from matchzoo import chain_transform, build_vocab_unit
+
+logger = logging.getLogger(__name__)
+tqdm.pandas()
+
+
+class MVLSTMPreprocessor(engine.BasePreprocessor):
+    """MVLSTMModel preprocessor."""
+
+    def __init__(self, fixed_length: list = [10, 10]):
+        """
+        MVLSTM Model preprocessor.
+
+        :param fixed_length: The fixed length of 'text_left' 
+            and 'text_right'.
+
+        Example:
+            >>> import matchzoo as mz
+            >>> train_data = mz.datasets.toy.load_train_classify_data()
+            >>> test_data = mz.datasets.toy.load_test_classify_data()
+            >>> mvlstm_preprocessor = mz.preprocessors.MVLSTMPreprocessor()
+            >>> train_data_processed = mvlstm_preprocessor.fit_transform(train_data)
+            >>> type(train_data_processed)
+            <class 'matchzoo.data_pack.data_pack.DataPack'>
+            >>> test_data_transformed = mvlstm_preprocessor.transform(test_data)
+            >>> type(test_data_transformed)
+            <class 'matchzoo.data_pack.data_pack.DataPack'>
+
+        """
+        super().__init__()
+        self._fixed_length = fixed_length
+        self._left_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length[0], pad_mode='post')
+        self._right_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length[1], pad_mode='post')
+        self._VocabularyUnit = processor_units.VocabularyUnit
+
+    def fit(self, data_pack: DataPack, verbose=1):
+        """
+        Fit pre-processing context for transformation.
+
+        :param verbose: Verbosity.
+        :param data_pack: data_pack to be preprocessed.
+        :return: class:`MVLSTMPreprocessor` instance.
+        """
+        units = self._default_processor_units()
+        data_pack = data_pack.apply_on_text(chain_transform(units),
+                                            verbose=verbose)
+        vocab_unit = build_vocab_unit(data_pack, verbose=verbose)
+
+        self._context.update(vocab_unit.state)
+        self._context['embedding_input_dim'] = len(
```

Comment:
Instead of add a `embedding_input_dim` and storing `vocab_unit.state`, simply do `self._context['vocab_unit'] = vocab_unit`.  `embedding_input_dim` can be calculated later using `len(preprocessor.context['vocab_unit'].state['term_index']) + 1`.